### PR TITLE
User story 27-28

### PR DESCRIPTION
**Pull request by adumortier**
- added a link `My Orders` to the `user` `profile` page 
- added information to the `Order` show page (`total_quantity`, `grandtotal`, `id`, `creation date`, `update date`, `order status`)
- added the test for the instance method `total_quantity` of an `order`
- added the instance method `total_quantity` to the `order` model

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -11,4 +11,8 @@ class Order <ApplicationRecord
   def grandtotal
     item_orders.sum('price * quantity')
   end
+
+  def total_quantity
+    items.sum(:quantity)
```
Comment (cgaddis36) [inline app/models/order.rb:16]: nice use of ActiveRecord

File: spec/features/profile/profile_spec.rb [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      fill_in :city, with: @city
+      fill_in :state, with: @state
+      fill_in :zip, with: @zip_code.to_i
+
+      click_button "Create Order"
+
+      visit "/profile"
+
+      click_link('My Orders')
+      expect(current_path).to eq("/profile/orders")
```
Comment (cgaddis36) [inline spec/features/profile/profile_spec.rb:76]: thorough testing for order link

File: spec/features/profile/profile_spec.rb [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+      visit "/cart"
+
+      click_on "Checkout"
+
+      fill_in :name, with: @username
+      fill_in :address, with: @street_address
+      fill_in :city, with: @city
+      fill_in :state, with: @state
+      fill_in :zip, with: @zip_code.to_i
```
Comment (cgaddis36) [inline spec/features/profile/profile_spec.rb:90]: not a huge fan of the "to_i" in the test but I assume we will just erase this once the tables have been changed to take in a string so nice job there

File: spec/features/profile/profile_spec.rb [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+       visit "/profile/orders"
+
+      within("div#order_#{order2.id}") do
+        expect(page).to have_content("Order id# = #{order2.id}")
+        expect(page).to have_content("Order made on: #{order2.created_at}")
+        expect(page).to have_content("Order updated on: #{order2.updated_at}")
+        
+        expect(page).to have_content("Total Quantity: #{order2.total_quantity}")
+        expect(page).to have_content("Grand Total: #{order2.grandtotal}")
```
Comment (cgaddis36) [inline spec/features/profile/profile_spec.rb:139]: Nice edge case testing to ensure we have info for both items on the order index

**Review by cgaddis36 — APPROVED** [review]

File: spec/features/profile/profile_spec.rb (hunk 3 above)
Comment (KellyIB) [reply spec/features/profile/profile_spec.rb:90]: I agree with Chase on this one, but when we convert the zip to strings, we can just tweak it.

**Review by KellyIB — APPROVED** [review]
Nice.  Looks clean to me.

**Pull request merged**